test/e2e/scale.go

```diff
@@ -0,0 +1,577 @@
+
+		// Setup a Namespace where to host objects for this spec and create a watcher for the namespace events.
+		namespace, cancelWatches = setupSpecNamespace(ctx, specName, input.BootstrapClusterProxy, input.ArtifactFolder)
+	})
+
+	It("Should create a workload cluster", func() {
+		infrastructureProvider := clusterctl.DefaultInfrastructureProvider
+		if input.InfrastructureProvider != nil {
+			infrastructureProvider = *input.InfrastructureProvider
+		}
+
+		flavor := clusterctl.DefaultFlavor
+		if input.Flavor != nil {
+			flavor = *input.Flavor
+		}
+
+		controlPlaneMachineCount := pointer.Int64(1)
+		if input.ControlPlaneMachineCount != nil {
+			controlPlaneMachineCount = input.ControlPlaneMachineCount
+		}
+		// If env variable is defined that will take precedence.
+		if input.E2EConfig.HasVariable(scaleControlPlaneMachineCount) {
+			controlPlaneMachineCountStr := input.E2EConfig.GetVariable(scaleControlPlaneMachineCount)
+			controlPlaneMachineCountInt, err := strconv.Atoi(controlPlaneMachineCountStr)
+			Expect(err).NotTo(HaveOccurred())
+			controlPlaneMachineCount = pointer.Int64(int64(controlPlaneMachineCountInt))
+		}
+
+		workerMachineCount := pointer.Int64(1)
+		if input.WorkerMachineCount != nil {
+			workerMachineCount = input.WorkerMachineCount
+		}
+		// If env variable is defined that will take precedence.
+		if input.E2EConfig.HasVariable(scaleWorkerMachineCount) {
+			workerMachineCountStr := input.E2EConfig.GetVariable(scaleWorkerMachineCount)
+			workerMachineCountInt, err := strconv.Atoi(workerMachineCountStr)
+			Expect(err).NotTo(HaveOccurred())
+			workerMachineCount = pointer.Int64(int64(workerMachineCountInt))
+		}
+
+		clusterCount := input.ClusterCount
+		// If env variable is defined that will take precedence.
+		if input.E2EConfig.HasVariable(scaleClusterCount) {
+			clusterCountStr := input.E2EConfig.GetVariable(scaleClusterCount)
+			var err error
+			clusterCount, err = strconv.Atoi(clusterCountStr)
+			Expect(err).NotTo(HaveOccurred(), "%q value should be integer", scaleClusterCount)
+		}
+
+		concurrency := 1
+		if input.Concurrency > 0 {
+			concurrency = input.Concurrency
+		}
+		// If env variable is defined that will take precedence.
+		if input.E2EConfig.HasVariable(scaleConcurrency) {
+			concurrencyStr := input.E2EConfig.GetVariable(scaleConcurrency)
+			var err error
+			concurrency, err = strconv.Atoi(concurrencyStr)
+			Expect(err).NotTo(HaveOccurred(), "%q value should be integer", scaleConcurrency)
+		}
+
+		// TODO(ykakarap): Follow-up: Add support for legacy cluster templates.
+
+		By("Create the ClusterClass to be used by all workload clusters")
+		// Generate the yaml once and use it as the base template instead of creating the YAML multiple times.
+		// The only thing that needs to be changes in the YAML is the Cluster name.
+		// Note: The YAML will contain the ClusterClass as well. Check if it is okay to have all the workers
```

Comment:
This is something clusterctl already takes care of if we delegate to it regenerating the cluster yaml many times

Reply:
> https://github.com/kubernetes-sigs/cluster-api/pull/8833#discussion_r1229616591

With the current performance of clusterctl we can only do the following:
* create 1 cluster normally
* generate the cluster template again (this will omit the ClusterClass stuff). Use the second cluster template for all cluster creations